### PR TITLE
More indentation fixes

### PR DESCRIPTION
**Pull request by gmile**

### CODE REVIEW

**Comment by whatyouhide** [comment]
Hey @gmile, another awesome catch :) Can you quickly search through the whole guide and see if there are more instances of this bad indentation? So that we fix all of them and forget about it 😄

**Comment by gmile** [comment]
Hey @whatyouhide, sure! I just did a pass on docs. Included here everything I could find, let me know if you find any issues with fixes I made

File: getting-started/basic-operators.markdown [hunk 1]
```diff
@@ -108,7 +107,9 @@ true
 
 The reason we can compare different data types is pragmatism. Sorting algorithms don't need to worry about different data types in order to sort. The overall sorting order is defined below:
 
-    number < atom < reference < functions < port < pid < tuple < maps < list < bitstring
+```
+number < atom < reference < functions < port < pid < tuple < maps < list < bitstring
+```
```
Comment (whatyouhide) [inline getting-started/basic-operators.markdown]: This change doesn't add anything IMO, we can leave the indented version here :)

File: getting-started/mix-otp/dependencies-and-umbrella-apps.markdown [hunk 2]
```diff
@@ -95,10 +95,12 @@ For this reason, Mix supports "umbrella projects." Umbrella projects allow you t
...
-    + kv_umbrella
-      + apps
-        + kv
-        + kv_server
+```
++ kv_umbrella
+  + apps
+    + kv
+    + kv_server
+```
```
Comment (whatyouhide) [inline getting-started/mix-otp/dependencies-and-umbrella-apps.markdown]: Same as above.

File: getting-started/mix-otp/dependencies-and-umbrella-apps.markdown [hunk 3]
```diff
@@ -253,10 +255,12 @@ Now Mix will guarantee the `:kv` application is started before `:kv_server` is s
...
```
Comment (whatyouhide) [inline getting-started/mix-otp/dependencies-and-umbrella-apps.markdown]: Same as above

File: getting-started/mix-otp/distributed-tasks-and-configuration.markdown [hunk 4]
```diff
@@ -34,16 +36,18 @@ $ iex --sname foo
 
 You can see now the prompt is slightly different and shows the node name followed by the computer name:
 
-    Interactive Elixir - press Ctrl+C to exit (type h() ENTER for help)
-    iex(foo@jv)1>
+```
+Interactive Elixir - press Ctrl+C to exit (type h() ENTER for help)
+iex(foo@jv)1>
+```
```
Comment (whatyouhide) [inline getting-started/mix-otp/distributed-tasks-and-configuration.markdown]: Same as above.

File: getting-started/mix-otp/docs-tests-and-with.markdown [hunk 5]
```diff
@@ -70,13 +70,15 @@ end
...
-         lib/kv_server/command.ex:11: KVServer.Command (module)
+```
+1) test doc at KVServer.Command.parse/1 (1) (KVServer.CommandTest)
+   test/kv_server/command_test.exs:3
+   Doctest failed
+   code: KVServer.Command.parse "CREATE shopping\r\n" === {:ok, {:create, "shopping"}}
+   lhs:  :not_implemented
+   stacktrace:
+     lib/kv_server/command.ex:11: KVServer.Command (module)
+```
```
Comment (whatyouhide) [inline getting-started/mix-otp/docs-tests-and-with.markdown]: Same as above.

**Comment by whatyouhide** [comment]
Hey @gmile, so actually on second thought there are at least a couple cases where the backticks work better than indentation, so yeah, let's go with backticks everywhere :) Ignore my other comments.

I will merge as soon as I have @josevalim's approval on this backticks vs indentation thing :)

**Comment by gmile** [comment]
@whatyouhide noted on keeping backticks in this PR, no problem.

> I will merge as soon as I have @josevalim's approval on this backticks vs indentation thing :)

Sure thing!

**Comment by whatyouhide** [comment]
Hey @gmile, I lost track of this PR, so sorry! Could you rebase on master? In the meantime I'm sure @josevalim will give his 👍 as well and we can merge :)

File: getting-started/keywords-and-maps.markdown [hunk 6]
```diff
@@ -45,9 +45,9 @@ iex> new_list[:a]
 
 Keyword lists are important because they have three special characteristics:
 
-  * Keys must be atoms.
-  * Keys are ordered, as specified by the developer.
-  * Keys can be given more than once.
+* Keys must be atoms.
```
Comment (josevalim) [inline getting-started/keywords-and-maps.markdown]: The correct in markdown is to indent bullets.

Comment (gmile) [inline getting-started/keywords-and-maps.markdown]: > The correct in markdown is to indent bullets.

@josevalim I couldn't find this in the md [spec](http://spec.commonmark.org/0.26/#list-items) for list items though.

This doesn't matter that much, I guess. I reverted this change.

**Comment by gmile** [comment]
@whatyouhide no worries at all! I just rebased this

**Comment by whatyouhide** [comment]
@josevalim is this shippable? I think so right?

**Comment by josevalim** [comment]
I am actually :-1: on most of those changes. Using triple backticks and spaces is mostly the same. And in many occasions where the space was removed, the space was there on purpose. For example, the reason why `test "foo"` was indented was because it is a snippet of a file (where it would be indented). Maybe doing so is a bad idea but it was certainly intentional.

Given this information, what are your thoughts @gmile and @whatyouhide ?

**Comment by whatyouhide** [comment]
@josevalim I see your concern and it was my concern initially as well. To move things forward, I think we can split this PR in two and the first part will be mergeable right away: the part where the code snippets that are indented 2 spaces (and are in triple backticks) are fixed to no indentation. Wdyt?

**Comment by josevalim** [comment]
> the part where the code snippets that are indented 2 spaces (and are in triple backticks) are fixed to no indentation

This is the part I said it was intentional. The indentation in the snippet mirrors the indentation we will have in the code once you move it to a file. Is that a good or bad idea? What are the pros and cons?

**Comment by whatyouhide** [comment]
@josevalim oh, sorry! I didn't get that 😄Makes sense now.

IMO, the pros is that the code won't look weird on the website by being indented and that you're not necessarily copy pasting this code into an editor. The cons is that if you are, it will be indented one level less than it should be, but in most editors this could be fixed by running the auto indentation on this snippet of code. Also I think it's worth mentioning that while this indentation has a reason to exist, I think it may still make some people ask themselves why it's this way, while no indentation would raise no questions.

**Comment by josevalim** [comment]
@whatyouhide it is your call. :)

**Comment by whatyouhide** [comment]
I think I'm 👍 on that, but maybe we can bring @lexmag in as well so we have a third (and hopefully decisive) opinion 😄

File: getting-started/where-to-go-next.markdown [hunk 7]
```diff
@@ -13,7 +13,9 @@ Eager to learn more? Keep reading!
 
 In order to get your first project started, Elixir ships with a build tool called Mix. You can get your new project started by simply running:
 
-    mix new path/to/new/project
+```bash
+mix new path/to/new/project
```
Comment (lexmag) [inline getting-started/where-to-go-next.markdown]: `$ mix new path/to/new/project` if we're going to use `bash` highlighting.

**Comment by lexmag** [comment]
Indented version looks a bit sloppy to me, I'd prefer not to indent. :bowtie:

File: getting-started/where-to-go-next.markdown (hunk 7 above)
Comment (whatyouhide) [inline getting-started/where-to-go-next.markdown]: Good point :)

**Comment by whatyouhide** [comment]
Awesome @lexmag, thanks for pitching in. Let's go with no indentation then, alright @gmile? :)

**Comment by gmile** [comment]
Thanks for opinions everyone!

Per discussion with @whatyouhide on Slack: I just pushed 2 more commits, I _hope_ I got closer to perfection this time :)

File: getting-started/mix-otp/introduction-to-mix.markdown [hunk 8]
```diff
@@ -190,17 +191,19 @@ assert 1 + 1 == 3
...
+     test/kv_test.exs:5
+     Assertion with == failed
+     code: 1 + 1 == 3
+     lhs:  2
+     rhs:  3
+     stacktrace:
+       test/kv_test.exs:6
+
+  Finished in 0.05 seconds (0.05s on load, 0.00s on tests)
+  1 test, 1 failure
```
Comment (whatyouhide) [inline getting-started/mix-otp/introduction-to-mix.markdown]: To be really, *really* pedantic, these two lines should not be indented cause they're not indented when you run `mix test` 😄

**Comment by whatyouhide** [comment]
@gmile left a last small comment, then we're definitely ready to ship!

**Comment by gmile** [comment]
@whatyouhide pushed a fix to that one :)

**Comment by whatyouhide** [comment]
Awesome, thanks so much @gmile, great job! 💟

**Pull request merged**